### PR TITLE
Ruby crud

### PR DESCRIPTION
**Pull request by OsBlaineOra**
These examples follow the Ruby OCI8 CRUD tutorial on my blog.

### CODE REVIEW

**Comment by cjbj** [comment]
- Files need some basic description; it's just too painful to follow links (which also may become dead).

- What about a cleanup SQL script to drop all the objects?

- Users will end up cutting/pasting from the code, so some more inline explanation of what is being shown (binding styles, committing, best practices, alternatives would be good)

- There are a LOT of files for simple examples. 

I'm looking forward to some more complex features being shown.  And, more importantly, Rails examples.

File: ruby/CRUD-examples/delete/extra_fun_1.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Code Sample from the tutorial at https://learncodeshare.net/2016/11/09/delete-crud-using-ruby-oci8/
+#  section titled "Extra Fun 1"
+# Using the base template, the example code executes a simple delete using named bind variables.
+
+require 'oci8'
+connectString = ENV['DB_CONNECT'] # The environment variable for the connect string: DB_CONNECT=user/password@database
+
+def get_all_rows(label, data_type = 'people')
+  connectString = ENV['DB_CONNECT'] # The environment variable for the connect string: DB_CONNECT=user/password@database
```
Comment (gvenzl) [inline ruby/CRUD-examples/delete/extra_fun_1.rb]: Why is the connect string retrieved twice? The location of the first retrieval indicates that the connect string is global but then it is retrieved again. Shouldn't either that function accept a connection as input parameter or the connect string be retrieved just before line 32 to make this clearer to the reader?

This applies multiple times throughout the code snippets.

**Comment by OsBlaineOra** [comment]
@cjbj 

1. Files need some basic description; it's just too painful to follow links (which also may become dead).
2. What about a cleanup SQL script to drop all the objects?
3. Users will end up cutting/pasting from the code, so some more inline explanation of what is being shown (binding styles, committing, best practices, alternatives would be good)
4.  There are a LOT of files for simple examples.

2 - It's already in there, see ruby/CRUD-examples/dropDBObjects.sql.
4 - I'm not sure what you would like me to change here?
1 & 3 - These examples are the Ruby version of [python/CRUD-examples](https://github.com/oracle/oracle-db-examples/tree/master/python/CRUD-examples), they are intended to be as similar as possible to those.  As with the Python examples (and the soon to be submitted Perl examples), the explanation details are on the tutorial page linked at the top of each file.

File: ruby/CRUD-examples/delete/extra_fun_1.rb (hunk 1 above)
Comment (OsBlaineOra) [inline ruby/CRUD-examples/delete/extra_fun_1.rb]: The get_all_rows functions do not have access to the top connectString variable.  I moved the top declaration down near where it's used.  I agree this looks cleaner.

**Comment by cjbj** [comment]
@OsBlaineOra 
- 1 &3 are pretty important to me.  I had/have no idea what the files were intended to show.
- 2 I must have missed it.  What about adding a short README.md file to the dir, pointing out the useful links and files?
- 4 was just a comment that we need moar and moar!

**Pull request merged**